app/src/main/java/com/asma/tasky/core/presentation/components/Navigation.kt

```diff
@@ -49,13 +56,64 @@ fun Navigation(
         }
 
         composable(Screen.TaskScreen.route) {
-            TaskScreen()
+            // todo check if a task object is passed from Agenda Screen
+            // todo also check if should show in edit mode
+
+            var title = stringResource(R.string.new_task)
+            var description = stringResource(R.string.task_description)
+
+            if (navController.currentBackStackEntry!!.savedStateHandle.contains(Constants.KEY_TITLE)) {
+                title =
+                    navController.currentBackStackEntry!!.savedStateHandle.get<String>(
+                        Constants.KEY_TITLE
+                    ) ?: stringResource(R.string.new_task)
+            }
+
+            if (navController.currentBackStackEntry!!.savedStateHandle.contains(Constants.KEY_DESCRIPTION)) {
+                description =
+                    navController.currentBackStackEntry!!.savedStateHandle.get<String>(
+                        Constants.KEY_DESCRIPTION
+                    ) ?: stringResource(R.string.task_description)
+            }
+
+            // default task if no task object passed
+            val editable = true
+            val task = Task(
+                title = title,
+                description = description,
+                id = "",
+                startDate = System.currentTimeMillis()
+            )
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                TaskScreen(task = task, editable, onEditTitle = {
+                    navController.navigate(Screen.EditTitleScreen.route + "/$it")
+                }, onEditDescription = {
+                    navController.navigate(Screen.EditDescriptionScreen.route + "/$it")
+                })
```

Comment:
And what if the device is running on a version lower than Oreo? Then that user can't use the screen or what? 😄